pkg/controller/command/verifiable/command.go

```diff
@@ -763,11 +761,43 @@ func prepareOpts(opts *ProofOptions, didDoc *did.Doc) (*ProofOptions, error) {
 		opts = &ProofOptions{}
 	}
 
+	opts.proofPurpose = "authentication"
+
+	if opts.PrivateKey != "" && opts.VerificationMethod == "" {
+		return nil, fmt.Errorf("verification method matching given private key is not provided")
+	}
+
+	authVMs := didDoc.VerificationMethods(did.Authentication)[did.Authentication]
```

Comment:
shouldn’t we look up by opts.proofPurpose?

Reply:
proofPurpose is no longer part of request opts. It is not exported. We default to authentication now.

Reply:
even if you hardcode opts.proofPurpose above, it is good to be generic :)